walnascar and whinlatter support

zboszor opened the pull request.

zboszor committed:
Add walnascar to LAYERSERIES_COMPAT

Yocto 5.2 Walnascar is close to being released. Support it.

Signed-off-by: Zoltán Böszörményi <zboszor@gmail.com>

zboszor committed:
mono (all versions): Add spaces around assignment operators

With:
https://lists.openembedded.org/g/bitbake-devel/message/17508
there are many WARNINGs from this layer

Signed-off-by: Zoltán Böszörményi <zboszor@gmail.com>

zboszor committed:
nuget: Add spaces around assignment operators

With:
https://lists.openembedded.org/g/bitbake-devel/message/17508
there are many WARNINGs from this layer.

Signed-off-by: Zoltán Böszörményi <zboszor@gmail.com>

zboszor committed:
msbuild: Add spaces around assignment operators

With:
https://lists.openembedded.org/g/bitbake-devel/message/17508
there are many WARNINGs from this layer

Signed-off-by: Zoltán Böszörményi <zboszor@gmail.com>

zboszor committed:
mono-xsp: Add spaces around assignment operators

With:
https://lists.openembedded.org/g/bitbake-devel/message/17508
there are many WARNINGs from this layer

Signed-off-by: Zoltán Böszörményi <zboszor@gmail.com>

zboszor committed:
python3-pythonnet: Add spaces around assignment operators

With:
https://lists.openembedded.org/g/bitbake-devel/message/17508
there are many WARNINGs from this layer

Signed-off-by: Zoltán Böszörményi <zboszor@gmail.com>

zboszor committed:
python3-clr-loader: Add spaces around assignment operators

With:
https://lists.openembedded.org/g/bitbake-devel/message/17508
there are many WARNINGs from this layer

Signed-off-by: Zoltán Böszörményi <zboszor@gmail.com>

ajlennon commented:
Seems to be failing

zboszor commented:
> Seems to be failing

Yes, but what? All CI tasks seem to be stuck in "queued" as far as I can see.

FWIW, I can start my build locally now, and there are no more `has a lack of whitespace around the assignment` warnings coming from `meta-mono` recipes.

ajlennon commented:
Yes I restarted the box and there's some very strange sstate error. Maybe I need to wipe out the sstate

ajlennon commented:
fwiw it is building here for some reason

https://github.com/DynamicDevices/meta-mono/actions/runs/14702446718

ajlennon commented:
Killed the sstate and re-ran. Something strange going on. First build works and second doesn't. Am wondering if the first build is polluting something?

https://github.com/DynamicDevices/meta-mono/actions/runs/14969532368/job/42060720344

zboszor committed:
layer-wide: Add whinlatter compatibility

Add whinlatter to LAYERSERIES_COMPAT in conf/layer.conf.

Add an abstract MONO_GIT_S setting in conf/layer.conf so recipes
that use git:// in SRC_URI can work in styhead, walnascar and
whinlatter, despite the different target directory for sources.

Use MONO_GIT_S in recipe that use git:// in SRC_URI.

Signed-off-by: Zoltán Böszörményi <zboszor@gmail.com>

zboszor commented:
I get this under current Yocto master:
```
| mkdir -p build-shared
| (cd build-shared && CC="gcc " CXX="g++ " "/mnt/zozo/yocto-5.3-arm/tmp-sicom/work/x86_64-linux/mono-native/6.12.0.206/recipe-sysroot-native/usr/bin/cmake" -D CMAKE_MAKE_PROGRAM="/mnt/zozo/yocto-5.3-arm/tmp-sicom/work/x86_64-linux/mono-native/6.12.0.206/recipe-sysroot-native/usr/bin/make" -D CMAKE_INSTALL_PREFIX:PATH="/mnt/zozo/yocto-5.3-arm/tmp-sicom/work/x86_64-linux/mono-native/6.12.0.206/recipe-sysroot-native/usr" -D BTLS_ROOT:PATH="/mnt/zozo/yocto-5.3-arm/tmp-sicom/work/x86_64-linux/mono-native/6.12.0.206/sources/mono-6.12.0.206/external/boringssl" -D SRC_DIR:PATH="/mnt/zozo/yocto-5.3-arm/tmp-sicom/work/x86_64-linux/mono-native/6.12.0.206/sources/mono-6.12.0.206/mono/btls" -D BTLS_CFLAGS:STRING="-isystem/mnt/zozo/yocto-5.3-arm/tmp-sicom/work/x86_64-linux/mono-native/6.12.0.206/recipe-sysroot-native/usr/include "  -DBTLS_ARCH="x86_64" -DBUILD_SHARED_LIBS=1 "/mnt/zozo/yocto-5.3-arm/tmp-sicom/work/x86_64-linux/mono-native/6.12.0.206/sources/mono-6.12.0.206/mono/btls")
| CMake Error at CMakeLists.txt:1 (cmake_minimum_required):
|   Compatibility with CMake < 3.5 has been removed from CMake.
| 
|   Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
|   to tell CMake that the project requires at least <min> but has been updated
|   to work with policies introduced by <max> or earlier.
| 
|   Or, add -DCMAKE_POLICY_VERSION_MINIMUM=3.5 to try configuring anyway.
| 
| 
| -- Configuring incomplete, errors occurred!
| make[3]: *** [Makefile:684: build-shared/Makefile] Error 1
```

I will have to patch the btls build so it may work with cmake 4.0 in Yocto master.

zboszor committed:
mono: Add recipe for version 6.14.1

This version compiles with cmake 4.0, i.e. under Yocto 5.3.

Signed-off-by: Zoltán Böszörményi <zboszor@gmail.com>

zboszor committed:
conf/layer.conf: Prefer 6.14.1 for mono and mono-native

Signed-off-by: Zoltán Böszörményi <zboszor@gmail.com>

zboszor committed:
ci: Extend the testing matrix to walnascar and master (whinlatter)

Signed-off-by: Zoltán Böszörményi <zboszor@gmail.com>

zboszor committed:
mono: Backport 3 patches from main to 6.14.1

Signed-off-by: Zoltán Böszörményi <zboszor@gmail.com>

zboszor committed:
Remove styhead from CI matrix

Not sure about the reason, but styhead vs arm and/or arm64
always gets stuck. Let's try without styhead.

Signed-off-by: Zoltán Böszörményi <zboszor@gmail.com>

ajlennon commented:
Been doing some maintenance trying to get things building correctly to pull in @spencercw's dotNet 10 contributions which seems to have worked out. Will take a look at what needs to be done here asap

zboszor committed:
conf/layer.conf: Add wrynose compatibility

Reverse the check for setting MONO_GIT_S, so it's future proof.
Hopefully.

Signed-off-by: Zoltán Böszörményi <zboszor@gmail.com>
